src/System.Device.Gpio/System/Device/Gpio/RaspberryBoardInfo.cs

```diff
@@ -0,0 +1,288 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+
+namespace System.Device.Gpio
+{
+    /// <summary>
+    /// Identification of Raspberry Pi board models
+    /// </summary>
+    internal class RaspberryBoardInfo
+    {
+        /// <summary>
+        /// The Raspberry Pi model.
+        /// </summary>
+        public enum Model
+        {
+            /// <summary>
+            /// Unknown model.
+            /// </summary>
+            Unknown,
+
+            /// <summary>
+            /// Raspberry Model A.
+            /// </summary>
+            RaspberryPiA,
+
+            /// <summary>
+            /// Model A+.
+            /// </summary>
+            RaspberryPiAPlus,
+
+            /// <summary>
+            /// Model B rev1.
+            /// </summary>
+            RaspberryPiBRev1,
+
+            /// <summary>
+            /// Model B rev2.
+            /// </summary>
+            RaspberryPiBRev2,
+
+            /// <summary>
+            /// Model B+.
+            /// </summary>
+            RaspberryPiBPlus,
+
+            /// <summary>
+            /// Compute module.
+            /// </summary>
+            RaspberryPiComputeModule,
+
+            /// <summary>
+            /// Pi 2 Model B.
+            /// </summary>
+            RaspberryPi2B,
+
+            /// <summary>
+            /// Pi Zero.
+            /// </summary>
+            RaspberryPiZero,
+
+            /// <summary>
+            /// Pi Zero W.
+            /// </summary>
+            RaspberryPiZeroW,
+
+            /// <summary>
+            /// Pi 3 Model B.
+            /// </summary>
+            RaspberryPi3B,
+
+            /// <summary>
+            /// Pi 3 Model B+.
+            /// </summary>
+            RaspberryPi3BPlus,
+
+            /// <summary>
+            /// Compute module 3.
+            /// </summary>
+            RaspberryPiComputeModule3,
+
+            /// <summary>
+            /// Pi 4 all versions.
+            /// </summary>
+            RaspberryPi4,
+        }
+
+        #region Fields
+
+        private readonly Dictionary<string, string> _settings;
+
+        private RaspberryBoardInfo(Dictionary<string, string> settings)
```

Comment:
what I had in mind is to not have dictionary at all but this is fine I guess since this is something we should be doing once per execution